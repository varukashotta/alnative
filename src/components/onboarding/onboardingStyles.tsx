import styled from 'styled-components/native';

interface IProps {
    backgroundColor: string;
}
export const Container = styled.View`
  background-color: ${(prop:IProps) => prop.backgroundColor};
  flex: 1;
`;
