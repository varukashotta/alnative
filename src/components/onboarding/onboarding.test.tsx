import {mount} from 'enzyme';
import React from 'react';
import Onboarding from '.';
import {FlatList, StatusBar} from 'react-native';
import Slide from './slide';
import {Container} from './onboardingStyles';

describe('Onboarding component', () => {
  let wrapper: any;

  let slider: any;

  let slideShallow: any;

  let slide: any;

  beforeEach(() => {
    wrapper = mount(
      <Onboarding
        slideButtons={true}
        data={[{heading: 'Heading', text: 'Content'}]}
        statusBar={true}
        _getStarted={jest.fn()}
      />,
    );

    slider = wrapper.find(FlatList);

    const navigate = jest.fn();

    slide = (
      <Slide
        _getStarted={jest.fn()}
        _navigateToPage={navigate}
        item={{}}
        showButtons={true}
        numberOfSlides={3}
        activeSlide={1}
      />
    );

    slideShallow = mount(slide);
  });

  it('should render without issues', () => {
    expect(wrapper.length).toBe(1);
  });

  it('should have prop navigateToProps', () => {
    expect(slideShallow.prop('_navigateToPage')).toBeDefined();
  });

  it('should have prop navigateToProps as function', () => {
    expect(
      typeof slideShallow.prop('_navigateToPage') === 'function',
    ).toBeTruthy();
  });

  describe('Get started button', () => {
    it('should have a getStarted prop', () => {
      expect(wrapper.prop('_getStarted')).toBeDefined();
    });

    it('should have a getStarted prop to be typeof function', () => {
      expect(typeof wrapper.prop('_getStarted') === 'function').toBeTruthy();
    });
  });

  it('should have Statusbar prop', () => {
    expect(wrapper.prop('statusBar')).toBeDefined();
  });

  it('should have Statusbar component defined', () => {
    expect(wrapper.contains(StatusBar)).toBeTruthy();
  });

  it('should have statusBar prop to be boolean', () => {
    expect(typeof wrapper.prop('statusBar') === 'boolean').toBeTruthy();
  });

  it('should have container wrapper as parent', () => {
    expect(wrapper.contains(Container)).toBeDefined();
  });

  it('should have data props', () => {
    expect(wrapper.prop('data')).toBeDefined();
  });

  it('should have data props that is an array', () => {
    expect(Array.isArray(wrapper.prop('data'))).toBeTruthy();
  });

  it('should have data array of objects', () => {
    expect(
      wrapper.prop('data').map((item) => typeof item === 'object'),
    ).toBeTruthy();
  });

  it('should have data prop with array objects that have a header if type string', () => {
    expect(
      wrapper
        .prop('data')
        .filter((item) => 'heading' in item && typeof item.heading === 'string')
        .length,
    ).toBeGreaterThanOrEqual(wrapper.prop('data').length);
  });

  it('should have data prop with array objects that have a text if type string', () => {
    expect(
      wrapper
        .prop('data')
        .filter((item) => 'text' in item && typeof item.heading === 'string')
        .length,
    ).toBeGreaterThanOrEqual(wrapper.prop('data').length);
  });

  it('should have numberOfSlides props', () => {
    expect(slideShallow.prop('numberOfSlides')).toBeDefined();
  });

  it('should have activeSlide props', () => {
    expect(slideShallow.prop('activeSlide')).toBeDefined();
  });

  describe('Slider Component Capabilities', () => {
    it('should have FlatList Component as child component', () => {
      expect(wrapper.find(FlatList).length).toEqual(1);
    });

    it('should have horizontal prop', () => {
      expect(slider.prop('horizontal')).toBeDefined();
    });

    it('should have horizontal prop that is equal to true', () => {
      expect(slider.prop('horizontal')).toBeTruthy();
    });

    it('should have data prop', () => {
      expect(slider.prop('data')).toBeDefined();
    });

    it('should have data prop that is an array', () => {
      expect(Array.isArray(slider.prop('data'))).toBeTruthy();
    });

    it('should have renderItem prop', () => {
      expect(slider.prop('renderItem')).toBeDefined();
    });

    it('should have renderItem prop to be function', () => {
      expect(slider.prop('renderItem')).toBeInstanceOf(Function);
    });

    it('should have renderItem prop to return Slide', () => {
      const mockFunction = jest.fn((point) => point);

      wrapper.find(FlatList).props('renderItem', mockFunction(slide));

      expect(mockFunction).toHaveBeenCalled();

      expect(mockFunction).toHaveReturnedWith(slide);
    });

    it('should have keyExtractor prop', () => {
      expect(slider.prop('keyExtractor')).toBeDefined();
    });

    it('should have keyExtractor prop to be function', () => {
      expect(slider.prop('keyExtractor')).toBeInstanceOf(Function);
    });
  });

  describe('Slide component capabilities', function () {
    it('should be defined', () => {
      expect(slideShallow.prop('item')).toBeDefined();
    });

    it('should have a prop item that is an object', () => {
      expect(
        typeof slideShallow.prop('item') === 'object' &&
          slideShallow.prop('item') !== null,
      ).toBeTruthy();
    });

    it('should have a prop showButtons', () => {
      expect(slideShallow.prop('showButtons')).toBeDefined();
    });

    it('should have a prop showButtons that is of boolean type', () => {
      expect(
        typeof slideShallow.prop('showButtons') === 'boolean',
      ).toBeTruthy();
    });
  });
});
