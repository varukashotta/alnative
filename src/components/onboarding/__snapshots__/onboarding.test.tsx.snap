// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding component should match snapshot 1`] = `
<View
  backgroundColor="red"
  style={
    Array [
      Object {
        "backgroundColor": "red",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "heading": "Heading",
          "text": "Content",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          backgroundColor="red"
          style={
            Array [
              Object {
                "backgroundColor": "red",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "height": "100%",
                "position": "relative",
                "width": 750,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignContent": "center",
                  "alignSelf": "center",
                  "justifyContent": "center",
                  "paddingBottom": 30,
                  "paddingLeft": 30,
                  "paddingRight": 30,
                  "paddingTop": 30,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 52.5,
                    "marginVertical": 37.5,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Heading
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "grey",
                    "fontSize": 25.5,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Content
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "lightseagreen",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingVertical": 22.5,
            },
          ]
        }
      >
        <Text
          onPress={[Function]}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 30,
                "fontWeight": "bold",
                "textAlign": "center",
              },
            ]
          }
        >
          Get Started
        </Text>
      </View>
    </View>
  </View>
</View>
`;
